pkg/chunker/file/file.go

```diff
@@ -68,36 +76,39 @@ func (c *File) Read(ctx context.Context) <-chan []byte {
 	go func(ch chan []byte) {
 		defer close(ch)
 
-		watcher, err := fsnotify.NewWatcher()
-		if err != nil {
-			log.Printf("failed to watch: %v\n", err)
-			return
-		}
-		// nolint: errcheck
-		defer watcher.Close()
+		var (
+			watcherEvents chan fsnotify.Event
+			watcherErrors chan error
+		)
 
-		if err = watcher.Add(filepath.Dir(filename)); err != nil {
-			log.Printf("failed to watch add: %v\n", err)
-			return
-		}
-		offset, err := c.source.Seek(0, io.SeekStart)
-		if err != nil {
-			log.Printf("failed to seek: %v\n", err)
-			return
+		if c.options.Follow {
+			watcher, err := fsnotify.NewWatcher()
+			if err != nil {
+				log.Printf("failed to watch: %v\n", err)
+				return
+			}
+			// nolint: errcheck
+			defer watcher.Close()
+
+			watcherEvents = watcher.Events
+			watcherErrors = watcher.Errors
+
+			if err = watcher.Add(filepath.Dir(filename)); err != nil {
+				log.Printf("failed to watch add: %v\n", err)
+				return
+			}
 		}
 
 		buf := make([]byte, c.options.Size)
 
 		for {
 			for {
-				n, err := c.source.ReadAt(buf, offset)
+				n, err := c.source.Read(buf)
```

Comment:
figured out `ReadAt()` was not that necessary, as there was a `.Seek()` which anyway resets file offset.

So now chunker doesn't `.Seek()` and uses regular `.Read()`, this way it will be easier to combine with any kind of "tail" implementation which sets seek offset before passing file to this method.

The way we call chunker is that we always separate `*os.File` instance for the duration of the API call.